pkg/repo/chartrepo_test.go

```diff
@@ -40,37 +40,22 @@ const (
 	testURL        = "http://example-charts.com"
 )
 
-func TestLoadChartRepository(t *testing.T) {
-	r, err := NewChartRepository(&Entry{
-		Name: testRepository,
-		URL:  testURL,
-	}, getter.All(&cli.EnvSettings{}))
-	if err != nil {
-		t.Errorf("Problem creating chart repository from %s: %v", testRepository, err)
-	}
+// loadFromDir a directory of charts archives (including sub-directories),
+// appending to the repositores ChartPath
+func loadFromDir(t *testing.T, r *ChartRepository, dir string) {
+	dirInfo, err := os.Stat(dir)
+	require.Nil(t, err)
+	require.True(t, dirInfo.IsDir())
 
-	if err := r.Load(); err != nil {
-		t.Errorf("Problem loading chart repository from %s: %v", testRepository, err)
-	}
-
-	paths := []string{
-		filepath.Join(testRepository, "frobnitz-1.2.3.tgz"),
-		filepath.Join(testRepository, "sprocket-1.1.0.tgz"),
-		filepath.Join(testRepository, "sprocket-1.2.0.tgz"),
-		filepath.Join(testRepository, "universe/zarthal-1.0.0.tgz"),
-	}
+	globArchives := func(pattern string) []string {
+		archives, err := filepath.Glob(filepath.Join(dir, pattern))
+		require.Nil(t, err)
 
-	if r.Config.Name != testRepository {
-		t.Errorf("Expected %s as Name but got %s", testRepository, r.Config.Name)
+		return archives
 	}
 
-	if !reflect.DeepEqual(r.ChartPaths, paths) {
-		t.Errorf("Expected %#v but got %#v\n", paths, r.ChartPaths)
-	}
-
-	if r.Config.URL != testURL {
-		t.Errorf("Expected url for chart repository to be %s but got %s", testURL, r.Config.URL)
-	}
+	r.ChartPaths = append(r.ChartPaths, globArchives("*.tgz")...)
+	r.ChartPaths = append(r.ChartPaths, globArchives("**/*.tgz")...)
 }
 
 func TestIndex(t *testing.T) {
```

Comment:
`TestIndex` which originally used the `Load(0` function I think needs to remain, as it is testing the `Index()` function (and no other test does)
So it now uses `loadFromDir()` as a test helper